### PR TITLE
feat: async await

### PR DESCRIPTION
**Pull request by hacdias**
More background about this effort: https://github.com/ipfs/js-ipfs/issues/1670.

**Depends** on https://github.com/libp2p/js-peer-id/pull/87

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -34,27 +34,14 @@ class PeerInfo {
...
-        return callback(err)
-      }
-
-      callback(null, new PeerInfo(id))
-    })
-    return
+PeerInfo.create = async (peerId) => {
+  if (typeof peerId === 'undefined') {
+    peerId = await PeerId.create()
+  } else if (peerId && typeof peerId.toJSON !== 'function') {
```
Comment (alanshaw) [inline src/index.js]: One for a separate PR but this should probably be `else if (!PeerId.isPeerId(peerId))`

**Comment by hacdias** [comment]
Done @alanshaw 😄

**Comment by hacdias** [comment]
@alanshaw should we re-do this or close?

**Comment by jacobheun** [comment]
v0.16.0 is out ✔️

**Pull request merged**